homeassistant/components/igloohome/sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Implementation of the sensor platform."""
+
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientError
+from igloohome_api import Api, ApiException
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, PlatformNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IgloohomeConfigEntry
+from .entity import IgloohomeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: IgloohomeConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensor entities."""
+
+    api = entry.runtime_data
+
+    try:
+        devices_response = await api.get_devices()
```

Comment:
it probably makes sense to do that step in __init__ `async_setup_entry` and store the result in `entry.runtime_data` so you don't have to get the devices for every platform. This will also help you satisfy ` test-before-setup`